src/lua/api-gateway/validation/oauth2/userProfileValidator.lua

```diff
@@ -222,7 +247,7 @@ function _M:validateUserProfile()
 
             local isValid, failureErrorCode, failureMessage = self:isProfileValid(cachingObj)
             if isValid == true then
-                self:storeProfileInCache(cacheLookupKey, cachingObj)
+                self:storeProfileInCache(cacheTokenLookupKey, cachingObj)
```

Comment:
Does this mean that for every successful validation (every request) we resave the profile in cache?

Reply:
Only if the oauth provider is called. Please review the logic